ghcjs-src/Miso/Subscription/Geolocation.hs

```diff
@@ -0,0 +1,83 @@
+{-# LANGUAGE RecordWildCards   #-}
+{-# LANGUAGE OverloadedStrings #-}
+-----------------------------------------------------------------------------
+-- |
+-- Module      :  Miso.Subscription.Geolocation
+-- Copyright   :  (C) 2016-2017 David M. Johnson
+-- License     :  BSD3-style (see the file LICENSE)
+-- Maintainer  :  David M. Johnson <djohnson.m@gmail.com>
+-- Stability   :  experimental
+-- Portability :  non-portable
+----------------------------------------------------------------------------
+module Miso.Subscription.Geolocation
+  ( geoSub
+  , defGeoOptions
+  , GeoOptions (..)
+  , Geo (..)
+  ) where
+
+import GHCJS.Foreign.Callback
+import GHCJS.Marshal
+import GHCJS.Types
+import JavaScript.Object
+import JavaScript.Object.Internal
+
+import Miso.Html.Internal         ( Sub )
+import Miso.String
+
+data GeoOptions = GeoOptions
+  { enableHighAccuracy :: Bool
+  , maximumAge :: Int
+  , timeout :: Int
+  } deriving (Show, Eq)
+
+instance ToJSVal GeoOptions where
+  toJSVal GeoOptions {..} = do
+    o@(Object j) <- create
+    toJSVal enableHighAccuracy >>= \e ->
+      setProp "enableHighAccuracy" e o
+    toJSVal maximumAge >>= \m ->
+      setProp "maximumAge" m o
+    toJSVal timeout >>= \t ->
+      setProp "timeout" t o
+    pure j
+
+defGeoOptions :: GeoOptions
+defGeoOptions = GeoOptions True 30000 27000
```

Comment:
The docs should probably mention what the defaults are. Otherwise the only way to figure this out is to look at the source.